src/SLCore.IntegrationTests/SLCoreTestRunner.cs

```diff
@@ -65,42 +65,51 @@ public void AddListener(ISLCoreListener listener)
-        noOpActiveSolutionBoundTracker.CurrentConfiguration.Returns(BindingConfiguration.Standalone);
-        var noOpConfigScopeUpdater = Substitute.For<IConfigScopeUpdater>();
-
-        slCoreHandle = new SLCoreHandle(new SLCoreRpcFactory(slCoreTestProcessFactory, slCoreLocator,
-                new SLCoreJsonRpcFactory(new RpcMethodNameTransformer()),
-                new RpcDebugger { LogFilePath = Path.Combine(privateFolder, "logrpc.txt") },
-                new SLCoreServiceProvider(new NoOpThreadHandler(), logger),
-                new SLCoreListenerSetUp(listenersToSetUp)),
-            constantsProvider,
-            foldersProvider,
-            connectionProvider,
-            jarProvider,
-            noOpActiveSolutionBoundTracker,
-            noOpConfigScopeUpdater,
-            new NoOpThreadHandler());
-        await slCoreHandle.InitializeAsync();
+        try
+        {
+            Environment.SetEnvironmentVariable("SONARLINT_LOG_RPC", "true", EnvironmentVariableTarget.Process);
+
+            var slCoreLocator = Substitute.For<ISLCoreLocator>();
+            slCoreLocator.LocateExecutable().Returns(new SLCoreLaunchParameters("cmd.exe", $"/c {DependencyLocator.SloopBatPath}"));
+
+            var constantsProvider = Substitute.For<ISLCoreConstantsProvider>();
+            constantsProvider.ClientConstants.Returns(new ClientConstantsDto("SLVS_Integration_Tests",
+                $"SLVS_Integration_Tests/{VersionHelper.SonarLintVersion}"));
+            constantsProvider.FeatureFlags.Returns(new FeatureFlagsDto(true, true, false, true, false, false, true));
+            constantsProvider.TelemetryConstants.Returns(new TelemetryClientConstantAttributesDto("slvs_integration_tests", "SLVS Integration Tests",
+                VersionHelper.SonarLintVersion, "16.0", new()));
+
+            var foldersProvider = Substitute.For<ISLCoreFoldersProvider>();
+            foldersProvider.GetWorkFolders().Returns(new SLCoreFolders(storageRoot, workDir, userHome));
+
+            var connectionProvider = Substitute.For<IServerConnectionsProvider>();
+            connectionProvider.GetServerConnections().Returns(new Dictionary<string, ServerConnectionConfiguration>());
+
+            var jarProvider = Substitute.For<ISLCoreEmbeddedPluginJarLocator>();
+            jarProvider.ListJarFiles().Returns(DependencyLocator.AnalyzerPlugins);
+
+            var noOpActiveSolutionBoundTracker = Substitute.For<IActiveSolutionBoundTracker>();
+            noOpActiveSolutionBoundTracker.CurrentConfiguration.Returns(BindingConfiguration.Standalone);
+            var noOpConfigScopeUpdater = Substitute.For<IConfigScopeUpdater>();
+
+            slCoreHandle = new SLCoreHandle(new SLCoreRpcFactory(slCoreTestProcessFactory, slCoreLocator,
+                    new SLCoreJsonRpcFactory(new RpcMethodNameTransformer()),
+                    new RpcDebugger(new FileSystem(), new DateTime(0)), //we pass datetime so we do not create extra file for each run
```

Comment:
only line changed

Reply:
why not create a constructor override instead of this hack with datetime?

Reply:
Also, the log file originally was saved in the folder with the test name, now it's overridden by each test, which is not good